test/lit/basic/cont-no-gc.wast

```diff
@@ -0,0 +1,29 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+
+;; Test that we can write a binary without erroring when using stack switching
+;; without GC. This verifies we can emit continuation types properly even when
+;; GC is disabled
+
+;; RUN: wasm-opt %s --enable-stack-switching --enable-reference-types --roundtrip -S -o - | filecheck %s
+
+(module
+ ;; CHECK:      (type $proc (func))
+ (type $proc (func))
+ ;; CHECK:      (type $cont (cont $proc))
+ (type $cont (cont $proc))
+
+ ;; CHECK:      (func $main
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (cont.new $cont
+ ;; CHECK-NEXT:    (ref.func $main)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $main
+  (drop
+   (cont.new $cont
+    (ref.func $main)
+   )
```

Comment:
Actually, this doesn't make any sense without typed function references. If the `ref.func` is just a `funcref`, then there's no way to validate that the continuation type has the the same parameters as the type of the function reference it is initialized with.

And we don't differentiate between typed-function-references and gc as features, so continuations without GC simply don't make sense.

Reply:
Wait, wasn't there an effort to avoid continuations from depending on GC?

Anyhow, I'm fine with making stack switching depend on GC, is that the right thing then?

Reply:
Yes, but they still depend on typed function references, and we don't differentiate between that and GC. So making stack switching depend on GC is the right thing for our purposes. We can revisit this in the future (probably by adding a separate typed-function-references feature again) if anyone really needs it.

Reply:
Ok, I opened https://github.com/WebAssembly/binaryen/pull/7830 and will close this.